src/graph/sampling/neighbor/neighbor.cc

```diff
@@ -104,6 +95,41 @@ HeteroSubgraph SampleNeighbors(
   return ret;
 }
 
+HeteroSubgraph SampleNeighbors(
```

Comment:
this implementation is not efficient on a large graph. you are remove the edges from the input graph. you should remove edges from the sampled subgraph.